src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php

```diff
@@ -27,11 +27,14 @@ public function __invoke(string $composerJsonPath) : Generator
 
     private function getFilePaths(array $sourceDirs, string $packageDir) : array
     {
+        $flattened = array_reduce($sourceDirs, function(array $sourceDirs, $sourceDir) {
+            return $sourceDirs + (is_array($sourceDir) ? $sourceDir : [$sourceDir]);
```

Comment:
What can `$sourceDir` be, if not `array`? Should an `array` cast be used instead?